src/AssetsPage/AssetsTable/AssetsTable.test.jsx

```diff
@@ -24,31 +28,335 @@ const defaultProps = {
+      {
+        sort: 'date_added',
+        direction: 'desc',
+      },
+      {
+        sort: 'date_added',
+        direction: 'asc',
+      },
+      {
+        sort: 'date_added',
+        direction: 'desc',
+      },
+    ];
+
+    const assertValues = [
+      {
+        buttonIndexToClick: 2,
+        buttonToClickText: 'Date Added',
+        onSortColumnParameter: 'date_added',
+        onSortDirectionParameter: 'asc',
+      },
+      {
+        buttonIndexToClick: 2,
+        buttonToClickText: 'Date Added',
+        onSortColumnParameter: 'date_added',
+        onSortDirectionParameter: 'desc',
+      },
+      {
+        buttonIndexToClick: 0,
+        buttonToClickText: 'Name',
+        onSortColumnParameter: 'display_name',
+        onSortDirectionParameter: 'desc',
+      },
+    ];
+
+    sortableAssetProps.forEach((sortableAssetProp, index) => {
+      it(`calls onSort function with ${sortableAssetProp.direction} click on ${sortableAssetProp.sort === 'date_added' ? 'same' : 'different'} column`, () => {
+        const sortableAssetParameters = {
+          ...defaultProps.assetsParameters,
+          ...sortableAssetProp,
+        };
+
+        const testValues = assertValues[index];
+
+        wrapper = mount(
+          <AssetsTable
+            {...defaultProps}
+            updateSort={updateSortSpy}
+            assetsParameters={sortableAssetParameters}
+          />,
+        );
+
+        const sortButtons = wrapper.find('button').filterWhere(button => (
+          button.find('span > span .fa-sort').exists() ||
+          button.find('span > span .fa-sort-desc').exists() ||
+          button.find('span > span .fa-sort-asc').exists()
+        ));
+
+        expect(sortButtons).toHaveLength(3);
+
+        expect(sortButtons.at(testValues.buttonIndexToClick).filterWhere(button => (
+          button.text().includes(testValues.buttonToClickText)
+        ))).toHaveLength(1);
+
+        sortButtons.at(testValues.buttonIndexToClick).simulate('click');
+
+        expect(updateSortSpy).toHaveBeenCalledTimes(1);
+        expect(updateSortSpy).toHaveBeenLastCalledWith(
+          testValues.onSortColumnParameter,
+          testValues.onSortDirectionParameter,
+        );
+      });
+    });
+  });
+  describe('modal', () => {
+    let modal;
+
+    beforeEach(() => {
+      wrapper = mount(
+        <AssetsTable
+          {...defaultProps}
+        />,
+      );
+      modal = wrapper.find('[role="dialog"]');
+    });
+    it('renders', () => {
+      expect(modal).toHaveLength(1);
+    });
+    it('is closed by default', () => {
+      expect(modal.hasClass('modal-open')).toEqual(false);
+      expect(wrapper.state('modalOpen')).toEqual(false);
+    });
+    it('opens when trash button clicked; modalOpen state is true', () => {
+      const trashButtons = wrapper.find('button').filterWhere(button => button.hasClass('fa-trash'));
+      expect(trashButtons).toHaveLength(defaultProps.assetsList.length);
```

Comment:
I should remove this condition, since it's already tested (it's the second test of the render suite).